pkg/controller/migplan/registry_health.go

```diff
@@ -84,60 +84,63 @@ func (r *registryHealth) run() {
 				return
 			}
 
-			srcClient, err := srcCluster.GetClient(r.hostClient)
-			if err != nil {
-				log.Trace(err)
-				return
-			}
-			destClient, err := destCluster.GetClient(r.hostClient)
-			if err != nil {
-				log.Trace(err)
-				return
-			}
+			if srcCluster.Status.IsReady() {
```

Comment:
it is a good practice in golang to have the least amount of indentation. 

I would refactor this as:
1. create a function that take migcluster as output and returnes if plan sure be enqueued or not
2. in the function, if cluster is not ready return immediately, if not process it. Code will be readable, least amount of indents and we can unit test this function

Reply:
Thanks for the pointers, added a common function `isRegistryPodUnhealthy`